src/test/java/io/netty/incubator/codec/http3/Http3ControlStreamOutboundHandlerTest.java

```diff
@@ -83,32 +99,91 @@ public void testGoAwayIdDecreaseWorks() throws Exception {
     public void testGoAwayIdIncreaseFails() throws Exception {
         // Let's mark the parent as inactive before we close as otherwise we will send a close frame.
         parent.close().get();
-        EmbeddedChannel channel = newStream(new Http3ControlStreamOutboundHandler(
-                true, settingsFrame, new ChannelInboundHandlerAdapter()));
-        assertTrue(channel.writeOutbound(new DefaultHttp3GoAwayFrame(4)));
-        ReferenceCountUtil.release(channel.readOutbound());
+        EmbeddedChannel channel = newStream(newHandler());
 
-        try {
-            channel.writeOutbound(new DefaultHttp3GoAwayFrame(8));
-            fail();
-        } catch (Exception e) {
-            assertException(Http3ErrorCode.H3_ID_ERROR, e);
+        if (server) {
+            writeValidFrame(channel, new DefaultHttp3GoAwayFrame(4));
+            writeInvalidFrame(Http3ErrorCode.H3_ID_ERROR, channel, new DefaultHttp3GoAwayFrame(8));
+        } else {
+            writeValidFrame(channel, new DefaultHttp3GoAwayFrame(1));
+            writeInvalidFrame(Http3ErrorCode.H3_ID_ERROR, channel, new DefaultHttp3GoAwayFrame(3));
         }
+
         assertFalse(channel.finish());
     }
 
     @Test
     public void testGoAwayIdUseInvalidId() throws Exception {
+        assumeThat(server, is(true));
         parent.close().get();
         // Let's mark the parent as inactive before we close as otherwise we will send a close frame.
-        EmbeddedChannel channel = newStream(new Http3ControlStreamOutboundHandler(
-                true, settingsFrame, new ChannelInboundHandlerAdapter()));
-        try {
-            channel.writeOutbound(new DefaultHttp3GoAwayFrame(2));
-            fail();
-        } catch (Exception e) {
-            assertException(Http3ErrorCode.H3_ID_ERROR, e);
-        }
+        EmbeddedChannel channel = newStream(newHandler());
+
+        writeInvalidFrame(Http3ErrorCode.H3_ID_ERROR, channel, new DefaultHttp3GoAwayFrame(2));
+
+        assertFalse(channel.finish());
+    }
+
+    @Test
+    public void testClientGoAwayIdIsLimitedByMaxPushId() throws Exception {
+        assumeThat(server, is(false));
+        parent.close().get();
+        // Let's mark the parent as inactive before we close as otherwise we will send a close frame.
+        EmbeddedChannel channel = newStream(newHandler());
+
+        writeValidFrame(channel, new DefaultHttp3MaxPushIdFrame(8));
+        writeInvalidFrame(Http3ErrorCode.H3_ID_ERROR, channel, new DefaultHttp3GoAwayFrame(9));
+
+        assertFalse(channel.finish());
+    }
+
+    @Test
+    public void testClientGoAwayIdIsNotLimitedWhenMaxPushIdIsNotSet() throws Exception {
```

Comment:
This test is actually interesting. It tells me that I may have misunderstood the validations that are required to be done for the ID in go_away. 

The [`Connection Shutdown` section](https://quicwg.org/base-drafts/draft-ietf-quic-http.html#connection-shutdown) in the spec says:

> An endpoint that is attempting to gracefully shut down a connection can send a GOAWAY frame with a value set to the maximum possible value (2^62-4 for servers, 2^62-1 for clients). 

Which suggests that we should perhaps not restrict the IDs that are sent/received in the go_away other than the validations specified in the spec:
- Ids are correct depending on client/server
- ID is not greater than the previous ID.

Apologies for the back-and-forth and the misdirection

Reply:
no problem, it was in two separate commits, so I just reverted them. Thanks for your time reviewing this